crates/starknet-server/tests/test_advancing_time.rs

```diff
@@ -0,0 +1,186 @@
+        let resp_create_block = devnet
+            .post_json("/create_block".into(), Body::from(json!({}).to_string()))
+            .await
+            .unwrap();
+        let resp_body_create_block = get_json_body(resp_create_block).await;
+        assert!(resp_body_create_block["block_timestamp"].as_u64() > Some(past_time));
+
+        // wait 1 second
+        thread::sleep(time::Duration::from_secs(1));
+
+        // check if after mint block_timestamp is greater than last block
+        devnet.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+        let resp_latest_block = devnet
+            .post_json("/get_latest_block".into(), Body::from(json!({}).to_string()))
+            .await
+            .unwrap();
+        let resp_body_latest_block = get_json_body(resp_latest_block).await;
+        assert!(
+            resp_body_latest_block["block_timestamp"].as_u64()
+                > resp_body_create_block["block_timestamp"].as_u64()
+        );
+    }
+
+    #[tokio::test]
+    async fn set_time_in_far_future() {
+        // set time and assert
+        let future_time: u64 = 3376684800;
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let set_time_body = Body::from(
+            json!({
+                "time": future_time
+            })
+            .to_string(),
+        );
+        let resp = devnet.post_json("/set_time".into(), set_time_body).await.unwrap();
+        let resp_body = get_json_body(resp).await;
+        assert_eq!(resp_body["block_timestamp"], future_time);
+
+        // create block and check if block_timestamp is less than future_time
+        let resp_create_block = devnet
+            .post_json("/create_block".into(), Body::from(json!({}).to_string()))
+            .await
+            .unwrap();
+        let resp_body_create_block = get_json_body(resp_create_block).await;
+        assert!(resp_body_create_block["block_timestamp"].as_u64() < Some(future_time));
+
+        // wait 1 second
+        thread::sleep(time::Duration::from_secs(1));
+
+        // check if after mint block_timestamp is greater than last block
+        devnet.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+        let resp_latest_block = devnet
+            .post_json("/get_latest_block".into(), Body::from(json!({}).to_string()))
+            .await
+            .unwrap();
+        let resp_body_latest_block = get_json_body(resp_latest_block).await;
+        assert!(
+            resp_body_latest_block["block_timestamp"].as_u64()
+                > resp_body_create_block["block_timestamp"].as_u64()
+        );
+    }
+
+    #[tokio::test]
+    async fn set_time_empty_body() {
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let set_time_body = Body::from(json!({}).to_string());
+        let result = devnet.post_json("/set_time".into(), set_time_body).await.unwrap();
+        assert_eq!(result.status(), 422);
```

Comment:
How did you choose 422?

Reply:
It was not a choice, this is returned and I'm asserting that in tests.

Reply:
We aren't controlling which status code we are returning?

Reply:
I think we can close this conversation, it is what it is.